### PR TITLE
"MP Connection Lost" hidden while in solo

### LINKED ISSUE #49: 'Multiplayer Connection Lost' error message pops up when songs start playing
**Describe the bug**
When songs start playing on FFR R^3 Air 1.2.1, an error message pops up that reads 'MP Error: Multiplayer Connection Lost'.

**Screenshots**

**Reproduction Steps**
Follow these steps to reproduce bug:
1. Load any song.
2. When the song finishes loading and starts playing, the error message should pop up.

**Reproduction Rate**
This bug occurs 100% of the time.

**Expected behaviour**
This error message should not pop up when songs start playing.

**Desktop (please complete the following information):**
 - Platform: Air
 - Version: 1.2.1

**Additional context or comments**
The second screenshot shows the disconnection and reconnection messages in the Multiplayer Lobby. They appear when these error messages pop up, and are also 100% reproducible.

These error messages also pop up in the following scenarios, and are 100% reproducible:

1. Quit a song to go to the results screen and either click 'Replay song' or press the keybinding for restarting songs.
2. Click 'Spectator Editor', 'Multiplayer Editor' or 'Editor' in the options menu.

### PR DESCRIPTION
**Pull request by dave936**
Adds checks to hide the error popup if the player is in solo. Resolves #49.

### CODE REVIEW

**Comment by G-flat** [comment]
Have you tried outputting the trace of the SFSEvent when onConnectionLost is called? I feel that this issue might be more complicated than that.

You could try by calling trace(ObjectUtil.print_r(e.params)) and see what is output, similar to how src/com/flashfla/utils/SmartFox.as handles onConnectionLost.

**Comment by Zageron** [comment]
Haven't looked at the code yet, but fyi I disconnect the player from multiplayer on game start in solo because an active multiplayer connection causes lag spikes on some machines.

File: src/com/flashfla/net/Multiplayer.as [hunk 1]
```diff
@@ -847,7 +849,8 @@ package com.flashfla.net {
 			connected = false;
 
 			eventConnection();
-			eventError("Multiplayer Connection Lost");
+			if (inSolo == false)
+				eventError("Multiplayer Connection Lost");
```
Comment (Zageron) [inline src/com/flashfla/net/Multiplayer.as:853]:
```suggestion
			if (inSolo == false)
			{
				eventError("Multiplayer Connection Lost");
			}
```

File: src/com/flashfla/net/Multiplayer.as [hunk 2]
```diff
@@ -160,7 +161,8 @@ package com.flashfla.net {
 			server.connect(serverAddress, serverPort);
 		}
 
-		public function disconnect():void {
+		public function disconnect(_inSolo:Boolean = false):void {
+			inSolo = _inSolo;
 			server.disconnect();
 		}
```
Comment (Zageron) [inline src/com/flashfla/net/Multiplayer.as:167]:
```suggestion
		public function disconnect(_inSolo:Boolean = false):void
		{
			inSolo = _inSolo;
			server.disconnect();
		}
```

File: src/game/GamePlay.as [hunk 3]
```diff
@@ -380,7 +380,7 @@ package game
 
             if (MultiplayerSingleton.getInstance().connection.connected && !MultiplayerSingleton.getInstance().isInRoom())
             {
-                MultiplayerSingleton.getInstance().connection.disconnect();
+                MultiplayerSingleton.getInstance().connection.disconnect(true);
```
Comment (Zageron) [inline src/game/GamePlay.as:383]:
```suggestion
                bool isInSoloMode = true;
                MultiplayerSingleton.getInstance().connection.disconnect(isInSoloMode);
```

Comment (Zageron) [reply src/game/GamePlay.as:383]: Using magic bools or magic numbers should be avoided, instead make this readable for what it does right where it is used.

**Comment by Zageron** [comment]
Since actual unintentional disconnections wont call this with a flag, it will still show the error. (Which is good.)

**Comment by Zageron** [comment]
@G-flat oops, sorry.

**Comment by Zageron** [comment]
Crap. This file get missed in the full auto-format.

**Comment by Zageron** [comment]
@dave936 Let me know if you need any assistance with resolving the issue.

**Comment by Zageron** [comment]
Rebased and revised in #102.

**Comment by Zageron** [comment]
@all-contributors please add @dave936 for code

**Pull request closed without merging**